src/commands/transact.ts

```diff
@@ -119,16 +125,14 @@ async function register() {
         if (!transactionResult) {
             outputChannel.appendLine('Failed to transact.');
             return;
-        }
-
-        if (typeof transactionResult === 'object') {
+        } else if (typeof transactionResult.data === 'object') {
             outputChannel.appendLine(JSON.stringify(transactionResult, null, 2));
             outputChannel.show();
             return;
+        } else {
+            outputChannel.appendLine(transactionResult.toString());
```

Comment:
In case the transaction failed it prints `[Object object]`. You need to replace this whole if-else block with
```
if (!transactionResult) {
    outputChannel.appendLine('Failed to transact.');
    outputChannel.show();
    return;
} else {
    outputChannel.appendLine(JSON.stringify(transactionResult, null, 2));
    outputChannel.show();
    return;
}
```

Reply:
Should we 'return' in else if (typeof transactionResult.data === 'string') branch?
It looks to me the correct one should be:
`if (!transactionResult) {
    outputChannel.appendLine('Failed to transact.');
    outputChannel.show();
    return;
} else if (typeof transactionResult.data === 'string') {
    outputChannel.appendLine(transactionResult.data);
    outputChannel.show();
} else {
    outputChannel.appendLine(JSON.stringify(transactionResult, null, 2));
    outputChannel.show();
    return;
}`

Otherwise the following code line will be dead.

Reply:
I honestly think there is no need for `return` to begin with, the end of the function will do this return anyway. As for your question, because all those branches are dependent only one of them will be executed at a time so it does not matter if we keep the return or not.

First branch runs only if `transactionResult` is null, or undefined.

The second branch will run only if the `transactionResult` has `data` field in the form of string

The third branch will run in all other cases

Reply:
Make sense. I overlooked the disposable function scope. I'll remove the returns.